CorePayments/src/main/java/com/paypal/android/corepayments/API.kt

```diff
@@ -33,15 +36,29 @@ class API internal constructor(
         return http.send(httpRequest)
     }
 
+    /**
+     * Retrieves the merchant's clientID either from the local cache, or via an HTTP request if not cached.
+     * @return Merchant clientID.
+     */
     @Throws(PayPalSDKError::class)
-    suspend fun getClientId(): String {
+    suspend fun fetchCachedOrRemoteClientID(): String {
+        configuration.accessToken?.let { accessToken ->
+            clientIDCache.get(accessToken)?.let { cachedClientID ->
+                return cachedClientID
+            }
+        }
+
         val apiRequest = APIRequest("v1/oauth2/token", HttpMethod.GET)
         val httpRequest =
             httpRequestFactory.createHttpRequestFromAPIRequest(apiRequest, configuration)
         val response = http.send(httpRequest)
         val correlationID = response.headers["Paypal-Debug-Id"]
         if (response.isSuccessful) {
-            return parseClientId(response.body, correlationID)
+            val clientID = parseClientId(response.body, correlationID)
+            configuration.accessToken?.let { accessToken ->
```

Comment:
all of the `configuration.accessToken?.let { ... }` will go away once #117 is merged right?

Reply:
Yes, exactly!